labman/db/study.py

```diff
@@ -77,25 +191,31 @@ def samples(self, term=None, limit=None):
         Returns
         -------
         list of str
+            Returns tube identifiers if the `specimen_id_column` has been set
+            (in Qiita), or alternatively returns the sample identifier.
         """
-        with sql_connection.TRN as TRN:
-            sql = """SELECT sample_id
-                     FROM qiita.study_sample
-                     WHERE study_id = %s {}
-                     ORDER BY sample_id"""
+        # this acts as the tube identifier
+        specimen_id_column = self.specimen_id_column
 
-            if term is not None:
-                sql = sql.format("AND LOWER(sample_id) LIKE %s")
-                # The resulting parameter for LIKE is of the form "%term%"
-                sql_args = [self.id, '%%%s%%' % term.lower()]
-            else:
-                sql = sql.format("")
-                sql_args = [self.id]
-
-            if limit is not None:
-                sql += " LIMIT %s"
-                sql_args.append(limit)
+        # an empty wildcard will match all samples i.e. when term is None
+        term = '%%' if term is None else '%%%s%%' % term.lower()
 
+        with sql_connection.TRN as TRN:
+            if specimen_id_column is None:
+                sql = """SELECT sample_id
+                         FROM qiita.study_sample
+                         WHERE study_id = %s AND LOWER(sample_id) LIKE %s
```

Comment:
Why is it preferable to get sample_id from qiita.study_sample rather than getting it from the qiita.sample_{studyid} table as is done if there is a specimen_id_column?  Is it better performing or something?  Just curious because I would think that it would be possible (although maybe not advisable? :) to use the current SQL for when there is specimen id in BOTH these cases by doing the below before formatting the sql ...

     if specimen_id_column is None:
            specimen_id_column = "sample_id"

Reply:
This is a great suggestion!